File: server/libs/web-support/src/main/java/com/sharetimer/web/support/dto/ErrorRes.java
package com.sharetimer.web.support.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Getter;
import lombok.RequiredArgsConstructor;

@Getter
@RequiredArgsConstructor
@Schema(name = "Error Response", description = "Error Response")
public class ErrorRes {

  @JsonProperty("statusCode")
  @Schema(name = "statusCode", description = "Status Code", example = "400")
  private final String statusCode;

  @JsonProperty("statusName")
  @Schema(name = "statusName", description = "Status Name", example = "ArgumentNotValid")
  private final String statusName;

  @JsonProperty("message")
  @Schema(name = "message", description = "Error Message", example = "Invalid Request.")
  private final String message;
}

File: server/libs/web-support/src/main/kotlin/com/sharetimer/web/support/dto/ErrorRes.kt
package com.sharetimer.web.support.dto

import com.fasterxml.jackson.annotation.JsonProperty
import io.swagger.v3.oas.annotations.media.Schema

@Schema(name = "Error Response", description = "Error Response")
data class ErrorRes(
    @field:JsonProperty("statusCode")
    @field:Schema(name = "statusCode", description = "Status Code", example = "400")
    val statusCode: String,
    @field:JsonProperty("statusName")
    @field:Schema(name = "statusName", description = "Status Name", example = "ArgumentNotValid")
    val statusName: String,
    @field:JsonProperty("message")
    @field:Schema(name = "message", description = "Error Message", example = "Invalid Request.")
    val message: String,
)
